### PR TITLE
[Documentation] Improve VFS article

### PR DESCRIPTION
**Pull request by AnErrupTion**
I've kind of rewrote the VFS article to be more clear, have the example code pieces more consistent, and also added a new section for formatting the drive. Some typos have also been fixed but there might also still be some.

### CODE REVIEW

File: Docs/articles/Kernel/VFS.md [hunk 1]
```diff
@@ -3,56 +3,73 @@
...
-This message is printed by the register method and it provides info about the file system.
+This message is printed by the RegisterVFS() method and it provides info about the partition.
+
+After our VFS has been initialized, we can use more interesting functions, let's go over some of them:
+
+## Format drive
+
+**Note**: You don't have to format your drive if you're debugging your Cosmos project with VMWare. The build will automatically add an already formatted FAT32 VMDK file for your convenience.
+
+You can format your drive with the Format() function, just like this:
```
Comment (MishaTY) [inline Docs/articles/Kernel/VFS.md:34]: Please add that Cosmos can only format the drive that already has FAT32.

Comment (AnErrupTion) [reply Docs/articles/Kernel/VFS.md:34]: Really? Well I didn't know that. I'll add that.

**Comment by Arawn-Davies** [comment]
Just wondering, why are you using the VFS methods instead of the plugged System.IO methods, kinda defeats the purpose of them 😂really I think the documentation should include the minimum needed VFS partitioning and initialization methods, then try and use System.IO where possible

**Comment by AnErrupTion** [comment]
The original article used them, and also why not?

**Comment by quajak** [comment]
I think its better to use the actual system.io methods.

**Review by Arawn-Davies — CHANGES_REQUESTED** [review]
> The original article used them, and also why not?

Because, as stated around many places, the articles were written a long time ago before System.IO was plugged. Plugging the .NET framework (along with assemblies like System, System.IO) has long been one of the goals of Cosmos, so we should be using them.

Please try and update the PR to use the System.IO methods as opposed to VFS methods.
This way, if developers want to implement their own custom FS like my Siaranite FS
 [Siaranite-Solutions](https://github.com/Siaranite-Solutions/SDFS) , we can just plug System.IO instead of having to go a level lower using VFS.

File: Docs/articles/Kernel/VFS.md [hunk 2]
```diff
@@ -71,36 +90,38 @@ foreach (var directoryEntry in directory_list)
...
 
-This one is more tricky,
-We need to get a directoryEntryList, find files in the list and print the content to the screen.
+This one is a bit more tricky. We'll need to use our list of DirectoryEntry, loop through all files and print their raw content to the standard output.
 
-of course, we start with geting the directory listing;
+Of course, we'll start with geting the list of DirectoryEntry:
 
 ```C#
 var directory_list = fs.GetDirectoryListing(@"0:\");
```
[the remark this answers is not in the document]
Comment (AnErrupTion) [reply Docs/articles/Kernel/VFS.md:98]: Has FileInfo been plugged yet or not?

**Comment by AnErrupTion** [comment]
Done, I modified the article.

**Comment by Arawn-Davies** [comment]
> Done, I modified the article.

cheers mate, we're just updating gocosmos.org's code highlighting issues but I'll get the PR reviewed after :)

File: Docs/articles/Kernel/VFS.md [hunk 3]
```diff
@@ -72,122 +91,86 @@ foreach (var directoryEntry in directory_list)
...
 
-## Write to a file
+## Write to file
 
-Now we will write to an existing file.  
-Writing to a file is almost the same as reading from a file.  
-Always remember, we should use try catch block.  
+Now we will write to an existing file.
+Writing to a file is almost the same as reading from a file.
+Always remember that we should put our code in a try catch block.
```
Comment (Arawn-Davies) [inline Docs/articles/Kernel/VFS.md:147]: Could we eliminate some of these try {} catch (Exception ex) {} blocks? 
Any C# user should know that if a null parameter is passed to a method (in this case a non-existent file or directory), it's going to throw an exception.

If you just add a reminder about that at the beginning of the article that will suffice, In practice and for code cleanliness, I don't think it's a good idea to add a try-catch block every time a System.IO method is called. From personal experience, it's better to write project-specific FS functions that can catch these exceptions, as an operating system kernel will be highly frequently accessing IO. Maybe add a note for that too :+1:

**Review by Arawn-Davies — CHANGES_REQUESTED** [review]
Aside from this wee bit, it's a great article, thanks! When changed I'll go ahead and merge

File: Docs/articles/Kernel/VFS.md (hunk 3 above)
Comment (quajak) [reply Docs/articles/Kernel/VFS.md:147]: The problem is that currently the fs is not very stable and very easy to use incorrectly, which means if you dont use try catch blocks, errors will just cause the kernel to crash, making it a lot harder to debug. So I think its fine if we for now encourage users to use try catch blocks often.

Comment (Arawn-Davies) [reply Docs/articles/Kernel/VFS.md:147]: that's fair enough then :)  I'll go ahead and merge

**Review by Arawn-Davies — APPROVED** [review]

**Comment by Arawn-Davies** [comment]
@AnErrupTion can you remerge master into your fork? Won't let me merge until it's up-to-date again

**Comment by AnErrupTion** [comment]
Done!

**Comment by Arawn-Davies** [comment]
@valentinbreiz I've reinstated auto-merge so this PR will merge as soon as @AnErrupTion updates his fork.

**Pull request merged**